packages/amplify-graphql-api-construct/API.md

```diff
@@ -0,0 +1,145 @@
+## API Report File for "@aws-amplify/graphql-api-construct-alpha"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { CfnApiKey } from 'aws-cdk-lib/aws-appsync';
+import { CfnDataSource } from 'aws-cdk-lib/aws-appsync';
+import { CfnFunctionConfiguration } from 'aws-cdk-lib/aws-appsync';
+import { CfnGraphQLApi } from 'aws-cdk-lib/aws-appsync';
+import { CfnGraphQLSchema } from 'aws-cdk-lib/aws-appsync';
+import { CfnPolicy } from 'aws-cdk-lib/aws-iam';
+import { CfnResolver } from 'aws-cdk-lib/aws-appsync';
+import { CfnResource } from 'aws-cdk-lib';
+import { CfnRole } from 'aws-cdk-lib/aws-iam';
+import { CfnTable } from 'aws-cdk-lib/aws-dynamodb';
+import { Construct } from 'constructs';
+import { Duration } from 'aws-cdk-lib';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { ResolverConfig } from '@aws-amplify/graphql-transformer-core';
+import { SchemaFile } from 'aws-cdk-lib/aws-appsync';
+import { TransformerPluginProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { TransformParameters } from '@aws-amplify/graphql-transformer-interfaces';
+
+// @public (undocumented)
+export class AmplifyGraphQlApi extends Construct {
+    constructor(scope: Construct, id: string, props: AmplifyGraphQlApiProps);
+    // (undocumented)
+    readonly env: string;
+    // (undocumented)
+    readonly resources: AmplifyGraphQlApiResources;
+}
+
+// @public (undocumented)
+export type AmplifyGraphQlApiProps = {
+    schema: AmplifyGraphQlApiSchema;
+    envOverride?: string;
```

Comment:
We attempt to pull it from the cdk context, and provide an override for use-cases like using a CDK pipeline.

Reply:
In that case I'd drop `Override`. Explicit prop taking precedence over contextual data is quite normal behavior for mechanisms like this.
I'd also avoid using shortcuts and acronyms in public APIs. I.e. this could be just `environment` property.

Reply:
I'll drop this entirely I think, per our discussion offline.